core/types/block.go

```diff
@@ -293,6 +298,9 @@ func CopyHeader(h *Header) *Header {
 func (b *Block) DecodeRLP(s *rlp.Stream) error {
 	var eb extblock
 	_, size, _ := s.Kind()
+	if size == 0 {
```

Comment:
@jbhurat It looks that this change is making some unit tests fail. What was the rationale for adding this check? Thanks!

Reply:
This was added to support for nil blocks in Piggyback messages that were sent along with `RoundChange` and `Preprepare` messages. The logic for Piggyback messages was updated, but this was not updated. I have pushed a commit to remove this